### PR TITLE
Don't clear big render target if all pixels will be replaced

### PR DESCRIPTION
**Pull request by cosmic-linden**

### CODE REVIEW

**Review by Geenz — COMMENTED** [review]
LGTM - I like the new API addition.

**Review by Geenz — APPROVED** [review]

**Comment by Geenz** [comment]
You may want to re-merge with develop and let CI re-run. Looks like Linux build isn't happy.

File: indra/newview/pipeline.cpp [hunk 1]
```diff
@@ -7476,7 +7478,7 @@ void LLPipeline::generateSMAABuffers(LLRenderTarget* src)
             LLGLSLShader& blend_weights_shader = gSMAABlendWeightsProgram[fsaa_quality];
 
             dest.bindTarget();
-            dest.clear(GL_COLOR_BUFFER_BIT);
+            dest.invalidate(GL_COLOR_BUFFER_BIT);
```
Comment (RyeMutt) [inline indra/newview/pipeline.cpp:7481]: This could have side effects on SMAA as the edge detection shaders utilize discard.

Comment (Geenz) [reply indra/newview/pipeline.cpp:7481]: That's actually a pretty good point - discards get weird when you don't have something clear.  Do you have any good scenes to test with?

**Comment by cosmic-linden** [comment]
@Geenz The Linux build is complaining about an autobuild issue with conflicting dependencies, so I've merged the code into develop for now.

**Pull request merged**